gluefactory/models/lines/pold2_extractor.py

```diff
@@ -303,10 +305,29 @@ def _forward(self, data: dict) -> torch.Tensor:
         binary_distance_map = self.process_distance_map(distance_map)
 
         # Apply two stage filter
-        return self.two_stage_filter(
-            points, binary_distance_map, distance_map, angle_map, indices_image
+        filtered_idx = self.two_stage_filter(points, binary_distance_map, distance_map, angle_map, indices_image)
+
+        # Apply NMS
+        if self.conf.nms:
+            # Get line end points
+            lines = points[filtered_idx]
+
+            # Append indices to lines - (N,2,3)
```

Comment:
What dimension is the 3 in (N,2,3) representing?

Reply:
I would prefer to see a comment explaining this in a bit more detail. This will help us to be faster later if we need to do adaptions